### PR TITLE
fix(storage): fix looping calls to processFile

### PR DESCRIPTION
**Pull request by thaddmt**
#### Description of changes
- Using a combination of `processFile` and `path` would cause the processFile to be called on loop
- Seems to be caused by the `setProcessKey` being called before the getting the `identityId` for use with the path callback
- Moved the call to setPRocessedKey` after the call to `fetchAuthSession` and that seems to resolve the issue
- Updated e2e test to use `path` with `processFile`
- fixes https://github.com/aws-amplify/amplify-ui/issues/5467

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-storage/src/components/StorageManager/utils/getInput.ts [hunk 1]
```diff
@@ -63,6 +58,11 @@ export const getInput = ({
       hasCallbackPath ? path({ identityId }) : path
     }${processedKey}`;
 
+    if (processFile) {
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageManager/utils/getInput.ts:61]: Do we also need to add a call to `onProcessFileSuccess` inside the conditional block on [line 47](https://github.com/aws-amplify/amplify-ui/pull/5477/files#diff-04342a0ffee2005ce1b867d03610c88d0b63ec3b73a9c475fef0f66ef92d2f4aR47) for `key` inputs?

Comment (thaddmt) [reply packages/react-storage/src/components/StorageManager/utils/getInput.ts:61]: Yep you're totally right

**Review by reesscot — APPROVED** [review]

File: packages/react-storage/src/components/StorageManager/utils/__tests__/getInput.spec.ts [hunk 2]
```diff
@@ -186,6 +189,33 @@ describe('getInput', () => {
...
+    });
+
+    await input();
+
+    const fetchAuthSessionCallOrder = fetchAuthSpy.mock.invocationCallOrder[0];
+    const onProcessFileSuccessCallORder =
+      onProcessFileSuccess.mock.invocationCallOrder[0];
+    expect(fetchAuthSessionCallOrder).toBeLessThan(
+      onProcessFileSuccessCallORder
+    );
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageManager/utils/__tests__/getInput.spec.ts:211]: 🙏

**Review by calebpollman — APPROVED** [review]
🚢

**Pull request merged**